uber/site_sections/emails.py

```diff
@@ -21,26 +21,31 @@ def sent(self, session, **params):
 
     def pending(self, session, message=''):
         automated_emails = []
-        last_job_completed = SendAllAutomatedEmailsJob.last_result.get('completed', False)
-        categories_results = SendAllAutomatedEmailsJob.last_result.get('categories', None)
+        last_run = EmailDaemonStatus.last_result_looks_valid(session)
+        categories_results = session.query(EmailDaemonCategoryResult).all()
 
         count_query = session.query(Email.ident, func.count(Email.ident)).group_by(Email.ident)
         sent_email_counts = {c[0]: c[1] for c in count_query.all()}
 
         for automated_email in AutomatedEmail.instances.values():
-            category_results = categories_results.get(automated_email.ident, None) if categories_results else None
-            unsent_because_unapproved = category_results.get('unsent_because_unapproved', 0) if category_results else 0
+            ident = automated_email.ident
+            unsent_because_unapproved = 0
+
+            if categories_results:
+                category_results = next((result for result in categories_results if result.ident == ident), None)
```

Comment:
I recommend rewriting this line since it's somewhat confusing.  It took me a couple of re-reads to realize you were using a generator comprehension and using that to check for the presence of a result.

I recommend replacing this line and the original

```python
categories_results = session.query(EmailDaemonCategoryResult).all()
```

line with the following:

```python
category_results = session.query(EmailDaemonCategoryResult).filter_by(ident=ident).first()
```

which I'm pretty sure will do the same thing, unless I've misunderstood this code.